### DISCUSSION TITLE
Parcel not hot-reloading on files outside of root directory

### DISCUSSION THREAD

**Discussion by BenjaminWaye** [parcel-bundler/parcel#6424] [Q&A] [answered]
Our app has imports that are coming from directories outside of the root directory. When changes happen in the directories outside the changes are not picked up and do not cause a hot reload. How is the watcher setup? Does it only detect changes in the root?

**Comment by mischnic** [comment, marked as answer]
> Does it only detect changes in the root

Yes, only inside the project root (the folder that contains `yarn.lock` or `.git`)

**Reply by BenjaminWaye** [reply]
So I would need to extend the watcher? https://github.com/parcel-bundler/watcher

**Reply by mischnic** [reply]
Related issues:
https://github.com/parcel-bundler/watcher/pull/32
https://github.com/parcel-bundler/parcel/issues/4332

But in your case, but in your case, you're probably doing `import "../../../../something";`?

**Reply by BenjaminWaye** [reply]
It seems this issue also discusses it: https://github.com/parcel-bundler/parcel/issues/4950

Is there a way to extend the watcher functionality through a plugin? Looking through the documentation I am not sure where I would put such an extension in the .parcelrc

Yes we are doing an import "../../../../something";

**Reply by mischnic** [reply]
> Is there a way to extend the watcher functionality through a plugin? Looking through the documentation I am not sure where I would put such an extension in the .parcelrc

No
